pkg/storaged/stratis/pool.jsx

```diff
@@ -64,6 +65,197 @@ const _ = cockpit.gettext;
 const fsys_min_size = 512 * 1024 * 1024;
 const fsys_round_size = 1024 * 1024;
 
+/* Abstractions over the r6 and r8 API revisions.
+ */
+
+function get_key_descriptions(pool) {
+    const result = [];
+
+    if (!pool.Encrypted)
+        return result;
+
+    if (client.stratis_interface_revision < 8) {
+        const val = pool.KeyDescription;
+        if (val[0] && val[1][0])
+            result.push({ slot: null, keydesc: val[1][1] });
+    } else {
+        let val = pool.KeyDescriptions;
+        if ("t" in val)
+            val = val.v; // XXX - change notifications drop the variant wrapping for some reason.
+        if (pool.MetadataVersion == 1) {
+            if (val[0] && val[1][0])
+                result.push({ slot: null, keydesc: val[1][1] });
+        } else if (pool.MetadataVersion == 2) {
+            for (const kd of val)
+                result.push({ slot: kd[0], keydesc: kd[1] });
+        }
+    }
+
+    return result;
+}
+
+function get_clevis_infos(pool) {
+    const result = [];
+
+    if (!pool.Encrypted)
+        return result;
+
+    if (client.stratis_interface_revision < 8) {
+        const val = pool.ClevisInfo;
+        if (val[0] && val[1][0]) {
+            if (val[1][1][0] == "tang") {
+                const config = JSON.parse(val[1][1][1]);
+                result.push({ slot: null, pin: "tang", url: config.url });
```

Comment:
These 6 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-21927-b100443d-20250808-061643-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)